### ISSUE TITLE
Installation

### ISSUE THREAD

**Issue by bepolymathe** [swhalemwo/obr-viz#14]
I am not a python (or even dev) user but I would like to test obvz. 

Since it's not a package yet I have a doubt about the installation (spacemacs in my case). I'm used to install with the following command (example with org-brain-export) but I think it won't work... I wouldn't want to break everything ;-)

`(org-brain-export :location (recipe :fetcher github :repo "Kungsgeten/org-brain-export"))
`

Could you give me an example of a functional configuration ?

**Comment by michaelsjackson** [comment]
You might check this, https://github.com/swhalemwo/obvz/issues/6, I have step by step commands what I did, up to compiling the .cpp code, and it works this way, if you repeat this process it should work there as well, I guess. Worth trying my friend, good luck.

**Comment by bepolymathe** [comment]
Oh... that answers my question, but it's a little scary :scream: 
I'll try anyway ;-)

**Comment by michaelsjackson** [comment]
No, the solutions are also always there, just read thru, try the given solutions there, hope it will work, if not, you can write your scenario, or error messages here, I am not an expert, but there is always a way.

**Comment by bepolymathe** [comment]
Yes I know, you have to try ;-) 
Now I have the following error message...

![image](https://user-images.githubusercontent.com/29939805/81109715-cda0a380-8f1a-11ea-83ca-7d0f959f981f.png)

**Comment by michaelsjackson** [comment]
This is now where the c++ part comes, you can recompile fruch_v3 in your machine, so getting a fresh .so, exactly for your computer.  Very easy, try those steps I have written.

**Comment by bepolymathe** [comment]
[Here](https://github.com/swhalemwo/obvz/issues/6#issuecomment-622429094) It works for the first order. However, it blocks on the second one because the package doesn't have the same name in dnf (fedora) as in apt apparently. 

![image](https://user-images.githubusercontent.com/29939805/81111024-ae0a7a80-8f1c-11ea-95b3-a3f87e83ad76.png)

**Comment by bepolymathe** [comment]
After installing this package 

`eigen3-devel-3.3.7-3.fc31.noarch
`
I have this message again...
![image](https://user-images.githubusercontent.com/29939805/81113258-3cccc680-8f20-11ea-96c4-17c7b402ba15.png)

**Comment by michaelsjackson** [comment]
Installed pybind:
`pip3 install pybind11
`
I tried compiling with eigen installation
`sudo apt install libeigen3-dev`

So I am guessing you installed pybind11 and libeigen3-dev successfully.

Then, switching into directory where frucht_v3.cpp is in your computer, from terminal.
In my case
`cd /home/mj/packages/obvz/layout_optim/cpp/`

Then type
`whereis eigen3`

Then, with its result, in above directory, type something like:
Came further using, added -I /usr/include/eigen3:
`g++ -O3 -Wall -shared -std=c++17 -fPIC python3 -m pybind11 --includes -I /usr/include/eigen3 ./frucht_v3.cpp -o ./frucht_v3.so`
because in my case whereis eigen3 showed as result: 
`/usr/include/eigen3`

If this works, you have a fresh .so file, copy this over the original one, two directories above.  If you can compile this .so you are kind of finished. Then it should work. Good luck.

**Comment by bepolymathe** [comment]
Thank you for that clarification. 

```
[08:04:35]  ~/Applications/obvz/layout_optim/cpp : pip3 install pybind11
Requirement already satisfied: pybind11 in /home/manu/.local/lib/python3.8/site-packages (2.5.0)
```
It's amazing because I don't have a "libeigen3-dev" package on Fedora. 

However I can go further. `whereis eigen3` return 

`eigen3: /usr/include/eigen3 /usr/share/eigen3
`
but if i type 

`g++ -O3 -Wall -shared -std=c++17 -fPIC python3 -m pybind11 --include -I /usr/include/eigen3 ./frucht_v3.cpp -o ./frucht_v3.so`

It return

```
g++: erreur: python3: Aucun fichier ou dossier de ce type
g++: erreur: pybind11: Aucun fichier ou dossier de ce type
g++: erreur: option « -m » non reconnue sur la ligne de commande
```

**Comment by michaelsjackson** [comment]
Do not type, do copy paste always, then modify, above stays `--includes`, you wrote `--include` , not sure if this is the reason, just saw this difference.

Copy paste in linux:  Mark text, somewhere else hit middle mouse button.

**Comment by bepolymathe** [comment]
hi

I change for "include" because it's been offered to me...

```
[09:30:28]  ~/Applications/obvz/layout_optim/cpp : g++ -O3 -Wall -shared -std=c++17 -fPIC python3 -m pybind11 --includes -I /usr/include/eigen3 ./frucht_v3.cpp -o ./frucht_v3.so
g++: erreur: python3: Aucun fichier ou dossier de ce type
g++: erreur: pybind11: Aucun fichier ou dossier de ce type
g++: erreur: option « -m » non reconnue sur la ligne de commande
g++: erreur: option « --includes » non reconnue sur la ligne de commande, aviez-vous l'intention d'utiliser « --include » ?

```

**Comment by michaelsjackson** [comment]
Maybe this first:
`dnf group install 'Development Tools'`

have it from: https://www.techinfected.net/2016/10/how-to-write-compile-run-c--program-fedora-linux-terminal.html

**Comment by bepolymathe** [comment]
Yes I already have that to install but I keep getting the same error message... :thinking:

**Comment by bepolymathe** [comment]
Maybe @swhalemwo have a idea ?

**Comment by michaelsjackson** [comment]
Well, I have no fedora, and I am not a linux expert, with all its variants, just google about compiling c++ in fedora, and with your error in Google, maybe you will find some ideas there.
I did nothing extra, but in ubuntu 16.04, but not sure what other stuff I installed in the past, maybe those help here as well, who knows.

**Comment by bepolymathe** [comment]
Ok I'm going to look on my side but it's true that I'm not in my comfort zone when it comes to compiling C++. 

Thanks for your help.

**Comment by michaelsjackson** [comment]
Somehow it seems it can not find pybind11 and python3, not sure why, here those did not complain, but google will help I hope, there is nothing to be uncomfortable, you do not do anything, just search which replies people gave to similar problems: google: `fedora linux c++ compiling pybind11 python3 your_error_message` , for example.

**Comment by bepolymathe** [comment]
Thanks, I wouldn't have thought to google it...
More seriously, it is probably an interesting tool, but it will require a package in the long run (this is suggested in the README). I'll say here if I've found a solution when I'll have time to really search and play with compilation errors.

**Comment by michaelsjackson** [comment]
Good luck, I hope you will sort this out soon, nothing hurries, but this tool is worth the effort, very useful.  Then one can forget other mind mapping or concept mapping tools, export as .svg, in case you want to beautify it even more, just edit in inkscape, for example.

**Comment by bepolymathe** [comment]
When I run the `obvz-start` command I get an error message about dbus.

`cd ~/Applications/obvz && /usr/bin/python3 obvz.py dbus force: exited abnormally with code 1. [2 times]`

**Comment by michaelsjackson** [comment]
So you could compile the .so file successfully?  If yes, what did you do differently, for historic reasons you could mention it here, as a help for future users.

For obvz, start org-visualization first, and even there, add a few entries, in case you started a fresh brain with zero entries, 5..10 entries maybe, then you can run obvz-start. Then hitting in org-brain visualization n, so obvz can show something.  Best is hit n a few times, just to be sure.  You should see something visualized there, if all worked correctly.

**Comment by bepolymathe** [comment]
No, I still haven't been able to compile. That's the error message I get when I try to do it.

**Comment by michaelsjackson** [comment]
But before you finished you should not start obvz-start, of course this will not work without  the .so file, as far as I know at least.  Ok, if you would use not type force but dot, it might work, but I am using always force graphs here.

**Comment by bepolymathe** [comment]
Well then, a little progress. I have successfully compiled with the following command which points to the version of python on my machine. 

`g++ -O3 -Wall -shared -std=c++17 -fPIC -I /usr/include/eigen3/ -I /usr/include/python3.8 ./frucht_v3.cpp -o ./frucht_v3.so`

I have the `frucht_v3.so` file but I keep getting the following error message if I run the obvz-start command 

![image](https://user-images.githubusercontent.com/29939805/81779814-c159af00-94f5-11ea-99b5-4cea505aed8d.png)

I feel like I've got too new a version of python? (but I need it for other programs). @swhalemwo
